src/project.coffee

```diff
@@ -126,9 +174,13 @@ class Project extends Model
     if projectPath?
       directory = if fs.isDirectorySync(projectPath) then projectPath else path.dirname(projectPath)
       @rootDirectory = new Directory(directory)
-      if @repo = GitRepository.open(directory, project: this)
-        @repo.refreshIndex()
-        @repo.refreshStatus()
+
+      # For now, use only the repositoryProviders with a sync API.
+      for provider in @repositoryProviders
+        if provider.createRepositorySync
```

Comment:
This could be collapsed to:

```coffee
break if @repo = provider.createRepositorySync?(@rootDirectory)
```

As a bonus, calling methods that might not exist with `?` is slightly more correct in CoffeeScript because it actually verifies that the property is a function.